windows-agent/internal/proservices/landscape/executor_test.go

```diff
@@ -223,6 +291,39 @@ func TestInstall(t *testing.T) {
 	}
 }
 
+//nolint:revive // Context goes after testing.T
+func mockRootfsFileServer(t *testing.T, ctx context.Context) string {
+	t.Helper()
+
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/goodfile", func(w http.ResponseWriter, r *http.Request) {}) // Return empty file
+	mux.HandleFunc("/badfile", func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprintf(w, "MOCK_ERROR")
+		if err != nil {
+			t.Logf("mockRootfsFileServer: could not write response: %v", err)
+		}
+	})
+	mux.HandleFunc("/badresponse", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNotFound)
+	})
+
+	lis, err := (&net.ListenConfig{}).Listen(ctx, "tcp", "localhost:")
+	require.NoError(t, err, "Setup: mockRootfsFileServer could not listen")
+
+	go func() {
+		//nolint:gosec // For the sake of simplicity and readability.
+		err := http.Serve(lis, mux)
+		if err != nil {
+			t.Logf("mockRootfsFileServer: serve error: %v", err)
+		}
+	}()
+
+	addr := "http://" + lis.Addr().String()
+	t.Logf("Serving on %s", addr)
+	return addr
```

Comment:
we don’t return a way to teardown the server, do we? (closing `lis` connection). I think we should, even if this is in tests.